pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -124,14 +125,14 @@ func (o createAuthInfoOptions) run() error {
 }
 
 // authInfo builds an AuthInfo object from the options
-func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
+func (o *createAuthInfoOptions) modifyAuthInfo(cmd *cobra.Command, existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
-		certPath := o.clientCertificate.Value()
-		if o.embedCertData.Value() {
+	if cmd.Flags().Changed(clientcmd.FlagCertFile) {
```

Comment:
Your asking can you find out if a flag is "Changed" progmatically using the short?  no.  If you change it via the short version on the command line, it will show changed here...